tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -662,84 +831,317 @@ TEST(RDFVary, VaryDisplay) // TEST instead of TEST_P because Display is single-t
-{
-
-}
+   EXPECT_DOUBLE_EQ(h1s["nominal"].GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2s["nominal"].GetMean(), 2.);
 
-TEST_P(RDFVary, DISABLED_VaryProfile2D)
-{
+   EXPECT_DOUBLE_EQ(h1s["x:0"].GetMean(), 0.);
+   EXPECT_DOUBLE_EQ(h2s["x:0"].GetMean(), 0.);
 
+   EXPECT_DOUBLE_EQ(h1s["x:1"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(h2s["x:1"].GetMean(), 4.5);
 }
 
-TEST_P(RDFVary, DISABLED_VaryReduce)
+TEST(RDFVary, VaryReduce)
 {
-
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Reduce([](int x, int y) { return x + y; }, "x", 0);
+   auto hs = VariationsFor(h);
+   EXPECT_EQ(*h, 45);
+   EXPECT_EQ(hs["nominal"], 45);
+   EXPECT_EQ(hs["x:0"], 35);
+   EXPECT_EQ(hs["x:1"], 55);
 }
 
-TEST_P(RDFVary, DISABLED_VaryReport)
+TEST_P(RDFVary, VaryReport)
 {
-
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Filter([](int x) { return x > 5; }, {"x"}, "before")
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Filter([](int x) { return x > 7; }, {"x"}, "after")
+               .Report();
+   testing::internal::CaptureStdout();
+   h->Print();
+   std::string output = testing::internal::GetCapturedStdout();
+   auto expOut = "before    : pass=4          all=10         -- eff=40.00 % cumulative eff=40.00 %\n"
+                 "after     : pass=2          all=4          -- eff=50.00 % cumulative eff=20.00 %\n";
+   EXPECT_STREQ(output.c_str(), expOut);
+   EXPECT_THROW(VariationsFor(h), std::logic_error);
 }
 
-TEST_P(RDFVary, DISABLED_VaryStats)
+TEST_P(RDFVary, VaryStats)
 {
-
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Stats<int>("x");
+   EXPECT_THROW(VariationsFor(h), std::logic_error); // not (yet) implemented
 }
 
-TEST_P(RDFVary, DISABLED_VaryStdDev)
+TEST_P(RDFVary, VaryStdDev)
 {
+   auto h = ROOT::RDataFrame(3)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .StdDev<int>("x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_DOUBLE_EQ(*h, 1.);
+   EXPECT_DOUBLE_EQ(hs["nominal"], 1.);
+   EXPECT_DOUBLE_EQ(hs["x:0"], 1.);
+   EXPECT_DOUBLE_EQ(hs["x:1"], 1.);
```

Comment:
we should use a variation expression that yields different results for different variations :grimacing:

Reply:
why was this marked as resolved?

Reply:
Now, I pushed the local changes: I saw that even `EXPECT_DOUBLE_EQ` might not correctly indicate that 2 doubles are the same. As a solution, I "designed" my variations so that the standard deviation is 0. and/or 1. I think this is a better comparison now.

Reply:
uhm, weird that with just 3 values we got values out different enough from the expected ones that `EXPECTED_DOUBLE_EQ` complained...but ok